File: FilmFInderApi/FilmFinderApi/Models/Watchlist.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace FilmFinderApi.Models
{
    public class Watchlist
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [BsonElement("watchlistId")]
        public string WatchlistId { get; set; }

        [BsonElement("movieId")]
        public string MovieId { get; set; }

        [BsonElement("movieName")]
        public string MovieName { get; set; }

        [BsonElement("uid")]
        public string Uid { get; set; }

        [BsonElement("userName")]
        public string UserName { get; set; }

        [BsonElement("userEmail")]
        public string UserEmail { get; set; }  // Ensure this matches the frontend model

        [BsonElement("isAddedToWatchlist")]
        public bool IsAddedToWatchlist { get; set; }
    }
}

File: Frontend/src/app/models/watchlist.ts
export interface Watchlist {
    watchlistId: string;  // Ensure this is included if it's expected
    movieId: string;
    movieName: string;
    uid: string;
    userName: string;
    userEmail: string;   // Ensure this matches the backend model
    isAddedToWatchlist: boolean;
    }